x/src/cargo.rs

```diff
@@ -86,39 +107,65 @@ impl Cargo {
         self
     }
 
+    // Passes extra environment variables to be passed to the executable.
+    pub fn env_additions<I, S>(&mut self, envs: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (S, S)>,
+        S: AsRef<OsStr>,
+        //        S: std::convert::AsRef<std::ffi::OsStr>,
+    {
+        self.inner.envs(envs);
+        self
+    }
+
     pub fn run(&mut self) -> Result<()> {
         self.inner.stdout(Stdio::inherit()).stderr(Stdio::inherit());
         self.do_run(true).map(|_| ())
     }
 
+    // This execution will in general not produce std out file output, but will
+    // instead return a byte Vec containing the execution's stdout messages.
```

Comment:
Doc comment should be something like:

```
/// Runs this command, capturing the standard output into a `Vec<u8>`.
```

Reply:
done, with a little more on the next line as well about logging diffs